graphstorm-processing/tests/test_dist_heterogenous_loader.py

```diff
@@ -0,0 +1,662 @@
+"""
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License").
+You may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+from typing import Any, Dict, List, Sequence, Tuple
+import json
+import os
+import shutil
+import tempfile
+
+from numpy.testing import assert_allclose
+from pyspark.sql import SparkSession, DataFrame
+import pyspark.sql.functions as F
+import pyarrow.parquet as pq
+import pytest
+
+from graphstorm_processing.graph_loaders.dist_heterogeneous_loader import (
+    DistHeterogeneousGraphLoader,
+    NODE_MAPPING_INT,
+    NODE_MAPPING_STR,
+)
+from graphstorm_processing.data_transformations.dist_label_loader import SplitRates
+from graphstorm_processing.config.label_config_base import NodeLabelConfig, EdgeLabelConfig
+from graphstorm_processing.config.config_parser import (
+    StructureConfig,
+    create_config_objects,
+    EdgeConfig,
+)
+from graphstorm_processing.config.config_conversion import GConstructConfigConverter
+from graphstorm_processing.constants import COLUMN_NAME, MIN_VALUE, MAX_VALUE, VALUE_COUNTS
+
+pytestmark = pytest.mark.usefixtures("spark")
+_ROOT = os.path.abspath(os.path.dirname(__file__))
+LABEL_COL = "label"
+NUM_DATAPOINTS = 10000
+
+
+@pytest.fixture(autouse=True, name="tempdir")
+def tempdir_fixture():
+    """Create temp dir for output files"""
+    tempdirectory = tempfile.mkdtemp(
+        prefix=os.path.join(_ROOT, "resources/test_output/"),
+    )
+    yield tempdirectory
+    shutil.rmtree(tempdirectory)
+
+
+@pytest.fixture(scope="function", name="data_configs_with_label")
+def data_configs_with_label_fixture():
+    """Create data configuration object that contain labels"""
+    config_path = os.path.join(_ROOT, "resources/small_heterogeneous_graph/gconstruct-config.json")
+
+    with open(config_path, "r", encoding="utf-8") as conf_file:
+        gconstruct_config = json.load(conf_file)
+        gsprocessing_config = GConstructConfigConverter().convert_to_gsprocessing(gconstruct_config)
+
+    data_configs_dict = create_config_objects(gsprocessing_config["graph"])
+
+    return data_configs_dict
+
+
+@pytest.fixture(scope="function", name="no_label_data_configs")
+def no_label_data_configs_fixture():
+    """Create data configuration object without labels"""
+    config_path = os.path.join(
+        _ROOT, "resources/small_heterogeneous_graph/gconstruct-no-labels-config.json"
```

Comment:
Where is gconstruct-no-labels-config.json stored?
Will you add it into graphstorm-processing/tests/?

Reply:
Yes, I'll add in a follow-up PR.